### PR TITLE
Add .isCompleted() to transaction

### PR DESCRIPTION
**Pull request by thekuom**
Adds .isCompleted() to a transactor, which just calls the underlying .isCompleted() from the transaction. I needed it because in my application, I pass a transaction around as part of a request and as a safeguard, I am rolling back the transaction. However, sometimes the transaction is already complete. This just harmlessly exposes whether I am allowed to commit/rollback or not.

### CODE REVIEW

**Comment by kibertoad** [comment]
@thekuom Thank you, this is really useful! Could you please add documentation for it at https://github.com/knex/documentation?

**Comment by thekuom** [comment]
Here you go: https://github.com/knex/documentation/pull/223

File: test/unit/knex.js [hunk 1]
```diff
@@ -414,6 +414,37 @@ describe('knex', () => {
...
+    const knex = Knex(sqliteConfig);
+    const trxProvider = knex.transactionProvider();
+    const trx = await trxProvider();
+
+    await trx.rollback();
+
+    const completed = trx.isCompleted();
+    expect(completed).to.be.true;
+  });
+
```
Comment (elhigu) [inline test/unit/knex.js]: looks like test is missing where that API is used inside trx handler function.

Like this:

```
knex.transaction(trx => {
   trx.isCompleted();  
});
````

Comment (thekuom) [reply test/unit/knex.js]: Done

**Comment by kibertoad** [comment]
Released in 0.19.1

**Pull request merged**